File: packages/mobile_flutter/lib/utils/format.dart
import 'package:intl/intl.dart';

final _currencyFormat = NumberFormat('#,###', 'ko_KR');

String formatCurrency(int amount) {
  return '₩${_currencyFormat.format(amount)}';
}

String formatDate(String dateStr) {
  final date = DateTime.parse(dateStr);
  return DateFormat('yyyy-MM-dd').format(date);
}

String formatTime(String dateStr) {
  final date = DateTime.parse(dateStr);
  return DateFormat('HH:mm').format(date);
}

String formatDateTime(String dateStr) {
  final date = DateTime.parse(dateStr);
  return DateFormat('yyyy-MM-dd HH:mm').format(date);
}

String formatDateGroup(String dateStr) {
  final date = DateTime.parse(dateStr);
  const days = ['월', '화', '수', '목', '금', '토', '일'];
  final dayOfWeek = days[date.weekday - 1];
  return '${date.year}년 ${date.month}월 ${date.day}일 ($dayOfWeek)';
}

String maskCardNumber(String cardNumber) {
  if (cardNumber.length < 4) return cardNumber;
  return '****-${cardNumber.substring(cardNumber.length - 4)}';
}

File: packages/mobile/src/utils/format.ts
export function formatCurrency(amount: number): string {
  return `₩${amount.toLocaleString('ko-KR')}`;
}

export function formatDate(dateStr: string): string {
  const date = new Date(dateStr);
  const year = date.getFullYear();
  const month = String(date.getMonth() + 1).padStart(2, '0');
  const day = String(date.getDate()).padStart(2, '0');
  return `${year}-${month}-${day}`;
}

export function formatTime(dateStr: string): string {
  const date = new Date(dateStr);
  const hours = String(date.getHours()).padStart(2, '0');
  const minutes = String(date.getMinutes()).padStart(2, '0');
  return `${hours}:${minutes}`;
}

export function formatDateTime(dateStr: string): string {
  return `${formatDate(dateStr)} ${formatTime(dateStr)}`;
}

export function formatDateGroup(dateStr: string): string {
  const date = new Date(dateStr);
  const days = ['일', '월', '화', '수', '목', '금', '토'];
  const year = date.getFullYear();
  const month = date.getMonth() + 1;
  const day = date.getDate();
  const dayOfWeek = days[date.getDay()];
  return `${year}년 ${month}월 ${day}일 (${dayOfWeek})`;
}

export function maskCardNumber(cardNumber: string): string {
  if (cardNumber.length < 4) return cardNumber;
  return `****-${cardNumber.slice(-4)}`;
}
